src/alcotest-engine/test.ml

```diff
@@ -239,5 +239,18 @@ let check_raises ?here ?pos msg exn f =
             Fmt.pf ppf "%t%a %s: expecting %s, got %s." (pp_location ?here ?pos)
               Pp.tag `Fail msg (Printexc.to_string exn) (Printexc.to_string e))
 
+let match_raises ?here ?pos msg exnp f =
+  show_assert msg;
+  match collect_exception f with
+  | None ->
+      check_err (fun ppf () ->
+          Fmt.pf ppf "%t%a %s: got nothing." (pp_location ?here ?pos) Pp.tag
+            `Fail msg)
+  | Some e ->
+      if exnp e then
```

Comment:
Shouldn't this be `if not (exnp e) then`? It's confusing that this isn't tested and semantic expectations of the boolean returned by the function aren't documented anywhere